sematic/db/queries.py

```diff
@@ -189,6 +190,30 @@ def save_run(run: Run) -> Run:
     """
     _assert_external_jobs_not_removed([run])
     with db().get_session() as session:
+        existing_run_future_state = (
+            session.query(Run.future_state).filter(Run.id == run.id).one_or_none()
+        )
+        existing_run_future_state = (
+            existing_run_future_state[0]
+            if existing_run_future_state is not None
+            else None
+        )
+        new_state = run.future_state
+        if isinstance(new_state, FutureState):
+            # despite run.future_state being typed as FutureState, it
+            # is sometimes a string.
+            new_state = new_state.name  # type: ignore
+
+        if (
+            existing_run_future_state != new_state
+            and not FutureState.is_allowed_transition(
```

Comment:
Always allow transition to the same state and simplify the condition?

Reply:
Other places where we have "is_allowed_transition" (ex: for resolutions) don't allow moving from one state to itself, and I thought it best to remain consistent